lib/matplotlib/axes/_axes.py

```diff
@@ -5120,15 +5120,24 @@ def imshow(self, X, cmap=None, norm=None, aspect=None,
 
         """
 
+        temp = np.asarray(X)
+        if temp.ndim == 3 and isinstance(norm, mcolors.BivariateNorm):
+                    temp = norm(temp)
+                    X = cmap(temp, alpha=self.get_alpha(), bytes=True)
+
         if not self._hold:
             self.cla()
 
-        if norm is not None and not isinstance(norm, mcolors.Normalize):
-            msg = "'norm' must be an instance of 'mcolors.Normalize'"
+        isNorm = isinstance(norm, (mcolors.Normalize, mcolors.BivariateNorm))
+        if norm is not None and not isNorm:
+            msg = "'norm' must be an instance of 'mcolors.Normalize' " \
+                  "or 'mcolors.BivariateNorm'"
```

Comment:
Sure, but I don't see a reason why BivariateNorm shouldn't be subclassing Norm as it has the same architecture as any other norm...take data (scaler or vector)->map to 1D lut value ->get color, and color ->lut->scaler/vector.